ecl/hthor/hthor.cpp

```diff
@@ -10129,9 +10177,12 @@ MAKEFACTORY(FilterProject);
 
 extern HTHOR_API IHThorActivity * createGroupDedupActivity(IAgentContext & _agent, unsigned _activityId, unsigned _subgraphId, IHThorDedupArg & arg, ThorActivityKind kind)
 {
-    if(arg.compareAll())
+    unsigned flags = arg.getFlags();
+    if(flags & DFcompareall)
         return new CHThorGroupDedupAllActivity(_agent, _activityId, _subgraphId, arg, kind);
-    else if (arg.keepLeft())
+    else if (flags & DFkeepbest)
+        return new CHThorGroupDedupKeepBestActivity(_agent, _activityId, _subgraphId, arg, kind, flags & DFkeepleft);
```

Comment:
See prev. question, re. whether better not to implement best functionality in existing activity.